tests/v2/validation/provisioning/rke2/cert_rotation_test.go

```diff
@@ -0,0 +1,169 @@
+	r.Run(name, func() {
+		r.Run("initial", func() {
+			testSession := session.NewSession(r.T())
+			defer testSession.Cleanup()
+
+			testSessionClient, err := r.client.WithSession(testSession)
+			require.NoError(r.T(), err)
+
+			clusterName := provisioning.AppendRandomString(fmt.Sprintf("%s-%s", r.clusterName, provider.Name))
+			generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
+			machinePoolConfig := provider.MachinePoolFunc(generatedPoolName, namespace)
+
+			machineConfigResp, err := machinepools.CreateMachineConfig(provider.MachineConfig, machinePoolConfig, testSessionClient)
+			require.NoError(r.T(), err)
+
+			machinePools := machinepools.RKEMachinePoolSetup(nodesAndRoles, machineConfigResp)
+
+			cluster := clusters.NewRKE2ClusterConfig(clusterName, namespace, "calico", credential.ID, kubeVersion, machinePools)
+			clusterResp, err := clusters.CreateRKE2Cluster(testSessionClient, cluster)
+			require.NoError(r.T(), err)
+
+			kubeProvisioningClient, err := r.client.GetKubeAPIProvisioningClient()
+			require.NoError(r.T(), err)
+
+			result, err := kubeProvisioningClient.Clusters("fleet-default").Watch(context.TODO(), metav1.ListOptions{
+				FieldSelector:  "metadata.name=" + cluster.ObjectMeta.Name,
+				TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+			})
+			require.NoError(r.T(), err)
+			checkFunc := clusters.IsProvisioningClusterReady
+
+			err = wait.WatchWait(result, checkFunc)
+			require.NoError(r.T(), err)
+			assert.Equal(r.T(), clusterName, clusterResp.ObjectMeta.Name)
+
+			cluster, err = r.client.Provisioning.Cluster.ByID(clusterResp.ID)
+			require.NoError(r.T(), err)
+			require.NotNil(r.T(), cluster.Status)
+
+			// rotate certs
+			require.NoError(r.T(), r.rotateCerts(clusterName, 1))
+			// rotate certs again
+			require.NoError(r.T(), r.rotateCerts(clusterName, 2))
+		})
+	})
+}
+
+func (r *V2ProvCertRotationTestSuite) TestCertRotation() {
+	for _, providerName := range r.config.Providers {
+		subSession := r.session.NewSession()
+
+		provider := CreateProvider(providerName)
+
+		client, err := r.client.WithSession(subSession)
+		require.NoError(r.T(), err)
+
+		cloudCredential, err := provider.CloudCredFunc(client)
+		require.NoError(r.T(), err)
+
+		for _, kubernetesVersion := range r.config.KubernetesVersions {
+			r.testCertRotation(provider, kubernetesVersion, r.config.NodesAndRoles, cloudCredential)
+		}
+
+		subSession.Cleanup()
+	}
+}
+
+func (r *V2ProvCertRotationTestSuite) rotateCerts(id string, generation int64) error {
+	kubeProvisioningClient, err := r.client.GetKubeAPIProvisioningClient()
+	require.NoError(r.T(), err)
+
+	cluster, err := kubeProvisioningClient.Clusters(namespace).Get(context.TODO(), id, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	cluster.Spec.RKEConfig.RotateCertificates = &rkev1.RotateCertificates{
+		Generation: generation,
+	}
+
+	cluster, err = kubeProvisioningClient.Clusters(namespace).Update(context.TODO(), cluster, metav1.UpdateOptions{})
```

Comment:
I did have a question. Is there a reason you are using the Kubernetes Provisioning Controller/Client rather than the rancher client to update the cluster?

Reply:
If there is no specific reason just leave it. I'm introducing a new Steve v1 client, and this would get updated on my end anyways, just thought I should ask first.

Reply:
I think it was because using the other client was very awkward for this use case IIRC

Reply:
Ok cool, no worries. A rule a thumb when updating resources we want to use the rancher client in our validation tests. But in this case it's no big deal as I'm updating tests with the new Steve client anyway. Thanks Jake!

Reply:
No prob! Thanks for reviewing!